sympy/logic/boolalg.py

```diff
@@ -125,7 +126,11 @@ def equals(self, other):
 
         if self.has(Relational) or other.has(Relational):
             raise NotImplementedError('handling of relationals')
-        return self.atoms() == other.atoms() and \
+        more = self.atoms()
+        less = other.atoms()
+        if len(more) < len(less):
```

Comment:
Checking atoms is not the right thing here because the atoms are not necessarily the logical symbols of the formula:
```
In [10]: (x & (y > 1) & (y > 2)).atoms()
Out[10]: {1, 2, x, y}
```

Reply:
This is a quick exit. If one set of atoms is not a subset of the other -- even including numerical literals -- can it be the same as the other?

Reply:
I don't see any relationship between atoms and equals:
```python
In [11]: e1 = (x & (y > 1) & (y > 0))

In [12]: e2 = (x & (y > 1) & (y > -1))

In [13]: e1
Out[13]: x ∧ y > 0 ∧ y > 1

In [14]: e2
Out[14]: x ∧ y > -1 ∧ y > 1

In [15]: e1.atoms()
Out[15]: {0, 1, x, y}

In [16]: e2.atoms()
Out[16]: {-1, 1, x, y}

In [17]: simplify(e1) == simplify(e2)
Out[17]: True
```
What exactly is `.equals` supposed to mean?

Reply:
BooleanFunction.equals does not currently deal with arguments that are Relationals; until it does, I am not sure that we have to worry about the distinction of atoms and symbols.
```
>>> e1.equals(e2)
Traceback (most recent call last):
...
NotImplementedError: handling of relationals
```
`a.equals(b)` means that `a` and `b` though not being equal, can be manipulated to such a form. If Relationals were allowed in the comparison, `e1.equals(e2)` would be true since they both reduce to `x & (y>1)`.

Reply:
> BooleanFunction.equals does not currently deal with arguments that are Relationals

What about other Booleans like Contains?

Restricting relationals does not make it reasonable to use atoms.

Why does `Boolean.equals` raise an error rather than returning None like Expr.equals?

Reply:
This is a method of BooleanFunction:
```python
>>> var('a b')
(a, b)
>>> Contains(x,Interval(a,b))
Contains(x, Interval(a, b))
>>> isinstance(_,BooleanFunction)
False
```
None could be raised. It might be better to let the user know that there is no current hope of evaluation happening because even though the request makes sense, it is a NotImplemented feature. This is similar to `solve` raising NotImplemented instead of returning an empty solution.

Reply:
```
In [7]: b = Contains(x, Interval(1, 2))

In [8]: c = Contains(x, Interval(2, 3))

In [9]: d = Contains(x, Interval(1, 3))

In [10]: b | c
Out[10]: x ∈ [1, 2] ∨ x ∈ [2, 3]

In [11]: d
Out[11]: x ∈ [1, 3]

In [12]: (b | c).equals(d)
Out[12]: False
```